### ISSUE TITLE
No space left for CI runner

### ISSUE THREAD

**Issue by nWEIdia** [pytorch/test-infra#5246]
Runner name: 'i-064f08c3b644e70f3'
 
e.g. https://github.com/pytorch/pytorch/actions/runs/9182174281/job/25252243760

**Comment by huydhn** [comment]
TLDR, these runners ran out of space because they could not clean up files under `/home/ec2-user/actions-runner/_work` directory that are owned by root while the GitHub self-hosted runner is running under ec2-user user.

I have taken one faulty runner `i-01720d373e8095015` out of the pool and keep it for further investigation.  Here are some initial results:

* The runner has a partition of 150GB, out of which 140GB has been used up or 94%
* Here are the top folders.  `/home/ec2-user/actions-runner/_work/_temp` is the biggest culprit.

```
101GB	/home/ec2-user
76GB	/home/ec2-user/actions-runner/_work
76GB	/home/ec2-user/actions-runner
67GB	/home/ec2-user/actions-runner/_work/_temp
30GB	/home/ec2-user/actions-runner/_work/_temp/_github_home
28GB	/home/ec2-user/actions-runner/_work/_temp/_github_home/.cache
18GB	/home/ec2-user/miniconda
```

* `/home/ec2-user/actions-runner/_work/_temp/_github_home/.cache` is owned by root.

```
ls -la /home/ec2-user/actions-runner/_work/_temp/_github_home/.cache
total 0
drwxr-xr-x 7 root     root     78 May 22 22:22 .
drwxr-xr-x 8 ec2-user ec2-user 94 May 23 00:54 ..
drwxr-xr-x 3 root     root     25 May 22 22:22 bazel
drwxr-xr-x 3 root     root     23 May 22 22:22 bazelisk
drwxr-xr-x 3 root     root     17 May 22 08:24 huggingface
drwxr-xr-x 5 root     root     52 May 22 08:29 pip
drwxr-xr-x 3 root     root     17 May 22 08:24 torch
```

The runner couldn't clean up these files.  This is the same issue as https://github.com/actions/runner/issues/529.  I can see the permission exception in diag logs:

```
[2024-05-23 00:28:28Z INFO TempDirectoryManager] Cleaning runner temp folder: /home/ec2-user/actions-runner/_work/_temp
[2024-05-23 00:28:32Z ERR  TempDirectoryManager] System.AggregateException: One or more errors occurred. (Access to the path '/home/ec2-user/actions-runner/_work/_temp/_github_home/.cache/bazel/_bazel_root/b11141dfc570d9008511e0192b3ffd82/server' is denied.)
 ---> System.UnauthorizedAccessException: Access to the path '/home/ec2-user/actions-runner/_work/_temp/_github_home/.cache/bazel/_bazel_root/b11141dfc570d9008511e0192b3ffd82/server' is denied.
 ---> System.IO.IOException: Permission denied
   --- End of inner exception stack trace ---
   at System.IO.Enumeration.FileSystemEnumerator`1.CreateDirectoryHandle(String path, Boolean ignoreNotFound)
   at System.IO.Enumeration.FileSystemEnumerator`1.Init()
   at System.IO.Enumeration.FileSystemEnumerable`1..ctor(String directory, FindTransform transform, EnumerationOptions options, Boolean isNormalized)
   at System.IO.Enumeration.FileSystemEnumerableFactory.FileSystemInfos(String directory, String expression, EnumerationOptions options, Boolean isNormalized)
```

**Comment by huydhn** [comment]
Because these files are owned by root, they are coming from (not a completed list):

* Nova build workflow https://github.com/pytorch/test-infra/issues/5091
* FBGEMM, for example https://github.com/pytorch/FBGEMM/blob/main/.github/workflows/fbgemm_gpu_ci_cuda.yml#L52

**Comment by huydhn** [comment]
@seemethere @jeanschmidt What would be the steps to introduce a post script to the runner .env file to run `sudo chown ec2-user:ec2-user` like what GitHub documents here https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/running-scripts-before-or-after-a-job?  It seems like the cleanest approach given the diverse nature of workflows from different repos.  The runner can then clean up these files by itself.

Are there better approaches that you have in mind?

**Comment by huydhn** [comment]
Also, we have always had this issue, but it surfaces more frequently today because of the capacity sev https://github.com/pytorch/pytorch/issues/126895 where there are less runners to choose from the pool

**Comment by huydhn** [comment]
The fix https://github.com/pytorch/test-infra/pull/5269 has been landed but I will need for it to roll out to prod to confirm.  So, let's keep this open a bit longer.

**Comment by huydhn** [comment]
Confirmed that the cleanup script is working correctly on one of the newly launched Linux runner ` i-0b5078979b6dfb379`
